gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/TaskManager.java

```diff
@@ -124,19 +127,39 @@ private LocalDateTime getProcessNow() {
     private void addFileToTask(Task task, String fileType, String objectKey, String fileUrl) {
         LOGGER.info("New file added to task {} with file type {} at URL {}", task.getTimestamp(), fileType, fileUrl);
         ProcessFile processFile = task.getProcessFile(fileType);
+        String fileName = FilenameUtils.getName(objectKey);
+        if (processFile.getProcessFileStatus().equals(ProcessFileStatus.NOT_PRESENT)) {
+            addFileEventToTask(task, FileEventType.AVAILABLE, fileType, fileName);
+        } else {
+            addFileEventToTask(task, FileEventType.UPDATED, fileType, fileName);
+        }
         processFile.setFileUrl(fileUrl);
         processFile.setProcessFileStatus(ProcessFileStatus.VALIDATED);
         processFile.setLastModificationDate(getProcessNow());
         processFile.setFileObjectKey(objectKey);
-        processFile.setFilename(FilenameUtils.getName(objectKey));
+        processFile.setFilename(fileName);
         if (task.getProcessFiles().stream().map(ProcessFile::getProcessFileStatus).allMatch(processFileStatus -> processFileStatus.equals(ProcessFileStatus.VALIDATED))) {
             LOGGER.info("Task {} is ready to run", task.getTimestamp());
             task.setStatus(TaskStatus.READY);
         }
     }
 
+    private void addFileEventToTask(Task task, FileEventType fileEventType, String fileType, String fileName) {
+        String message = getFileEventMessage(fileEventType, fileType, fileName);
+        ProcessEvent event = new ProcessEvent(task, getProcessNow(), FILE_EVENT_DEFAULT_LEVEL, message);
+        task.getProcessEvents().add(event);
+    }
+
+    private String getFileEventMessage(FileEventType fileEventType, String fileType, String fileName) {
+        if (!fileEventType.equals(FileEventType.UPDATED)) {
+            return String.format("The %s : '%s' is %s", fileType, fileName, fileEventType.toString().toLowerCase());
+        } else {
+            return String.format("A new version of %s  is available : '%s'", fileType, fileName);
+        }
+    }
+
     public TaskDto getTaskDto(LocalDateTime timestamp) {
-        return taskRepository.findByTimestamp(timestamp).map(Task::createDtofromEntity).orElse(getEmptyTask(timestamp));
+        return taskRepository.findByTimestamp(timestamp).map(Task::createDtoFromEntity).orElse(getEmptyTask(timestamp));
```

Comment:
Can you check why there is a change here ?

Reply:
It was on purpose, not related to the feature in the PR though. It's a camel case fix

Reply:
My bad !